pkg/controller/hyperconverged/hyperconverged_controller.go

```diff
@@ -580,6 +584,15 @@ func (r *ReconcileHyperConverged) ensureCDI(instance *hcov1alpha1.HyperConverged
 		}
 	}
 
+	if !reflect.DeepEqual(found.Spec, cdi.Spec) {
+		if found.Spec.UninstallStrategy == nil {
```

Comment:
What if `found.Spec.UninstallStrategy` is not nil but some other invalid value that the user inserted manually into the CR spec? Do we have a validator there?

Reply:
No, we don't. But I think we can only respect the user provided value that should be eventually detected on CDI operator side.
A nil value is instead different because we know that in the past we didn't set any value and so we want to upgrade it to the current default which is not nil for new deployments.

Reply:
> No, we don't. But I think we can only respect the user provided value that should be eventually detected on CDI operator side.
> A nil value is instead different because we know that in the past we didn't set any value and so we want to upgrade it to the current default which is not nil for new deployments.

So we can't reconcile to default value if, for example, the user had a typo in setting this field in the CR.
Maybe we can check if the existing value is matching one of the permitted values of `UninstallStrategy`, otherwise - reconcile to default.
WDYT?